File: Learn/DataStructure/数据结构与算法之美/src/ConsoleApp1/ConsoleApp1/ArrayStack.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ConsoleApp1
{
    class ArrayStack<T>
    {
        private T[] items;
        private int count;
        private int n;

        public ArrayStack(int n)
        {
            this.items = new T[n];
            this.n = n;
            this.count = 0;
        }

        public bool Push(T item)
        {
            if (count == n) return false;
            items[count] = item;
            ++count;
            return true;
        }

        public T Pop()
        {
            if (count == 0) return default;
            T item = items[count-1];
            --count;
            return item;
        }
    }
}

File: Learn/DataStructure/数据结构与算法之美/src/idea/src/main/java/com/xjsaber/learn/idea/stack/ArrayStack.java
package com.xjsaber.learn.idea.stack;

public class ArrayStack {
    // 数组
    private String[] item;
    // 栈中元素个数
    private int count;
    // 栈的大小
    private int n;

    public ArrayStack(int n){
        this.item = new String[n];
        this.count = 0;
        this.n = n;
    }

    public boolean push(String value){
        // 栈满
        if (count == n) return false;
        this.item[count] = value;
        count++;
        return true;
    }

    public String pop(){
        // 栈空
        if (count == 0) return "";
        String value = this.item[count-1];
        count--;
        return value;
    }
}
